test/protocols/binary_test.exs

```diff
@@ -1,80 +1,183 @@
 defmodule BinaryProtocolTest do
+  use ThriftTestCase, gen_erl: true
   use ExUnit.Case
 
-  @thrift_file_path "./test/fixtures/app/thrift/simple.thrift"
+  @moduletag :integration
 
-  import ParserUtils
+  def round_trip_struct(data, serializer_mf, deserializer_mf) do
+    {serializer_mod, serializer_fn} = serializer_mf
+    {deserializer_mod, deserializer_fn} = deserializer_mf
 
-  setup_all do
-    @thrift_file_path
-    |> parse_thrift
-    |> compile_module
+    serialized = :erlang.apply(serializer_mod, serializer_fn, [:struct, data])
+    |> IO.iodata_to_binary
 
-    :ok
+    :erlang.apply(deserializer_mod, deserializer_fn, [serialized])
   end
 
-  test "it should not encode empty fields" do
-    assert <<0>> == serialize_user(user(:elixir))
+  @thrift_file name: "enums.thrift", contents: """
+  enum Status {
+    ACTIVE,
+    INACTIVE,
+    BANNED = 6,
+    EVIL = 0x20,
+  }
+
+  struct StructWithEnum {
+    1: Status status
+  }
+  """
+  thrift_test "encoding enums" do
+    encoder = {StructWithEnum.BinaryProtocol, :serialize}
+    decoder = {Erlang.Enums, :deserialize_struct_with_enum}
```

Comment:
I struggled with namespacing already a little bit (and of tests with user.ex defined stomping over each other. There's some inflexibility about having each file define a single module unless you obey default namespaces. Aside from good taste, there should be nothing preventing us from having `core.User` and `common.User`, but the default generator won't allow this.